src/helpers/dom/element.js

```diff
@@ -762,11 +762,47 @@ export function getComputedStyle(element, rootWindow = window) {
 /**
  * Returns the element's outer width.
  *
+ * @see {preciseOuterWidth}
  * @param {HTMLElement} element An element to get the width from.
  * @returns {number} Element's outer width.
  */
 export function outerWidth(element) {
-  return Math.ceil(element.getBoundingClientRect().width);
+  return element.offsetWidth;
+}
+
+/**
+ * Returns the element's outer width rounded up to the nearest integer,
+ * accounting for any scale transforms on parent elements. This function is
+ * much more costly than `outerWidth`, use sparingly.
+ *
+ * @param {HTMLElement} element An element to get the width from.
+ * @returns {number} Element's outer width.
+ */
+export function preciseOuterWidth(element) {
```

Comment:
Can't be some factors counted just once for full Handsontable, saved within property and updated only when necessary?

Reply:
`preciseOuterWidth` seems to only be executed on initial render, but many times, adding around `2ms` before the initial render on my machine. It would be more performant if we cached it, but I'm not really sure at which point should the cache get invalidated.

Reply:
Maybe just take out creating DOM element from method `preciseOuterWidth` for calculation (do it just once, ie. in `getWidths` above `forEach` - btw. please closure it within private method) and handle `scaleX` as method argument?

Reply:
You can alternatively count it just once for created `GhostTable` instance, but maybe some listener for translation change would be needed.

Reply:
Ah yea, it makes sense to just move it outside the for loop when the argument is the same. Didn't notice that before for some reason https://github.com/handsontable/handsontable/pull/7496/commits/ee77e2169ae2c1c29500e7cf7bc9e506bf55ee92

Reply:
Update, seems like doing this broke some tests (https://github.com/handsontable/handsontable/pull/7496/commits/ee77e2169ae2c1c29500e7cf7bc9e506bf55ee92), reverted in https://github.com/handsontable/handsontable/pull/7496/commits/ebcb871e94e9744818010dcb54aa6b1bae82c480

Reply:
Thus, I suggest moving and creating DOM element to another method which will return `scaleX`. It should be the same for every element in loop. This constant could be passed as second argument for `preciseOuterWidth` method.

Reply:
https://github.com/handsontable/handsontable/pull/7496/commits/e8e2b393a376836177944e5915237a8f6f30443e fingers crossed the tests pass